src/wng_building/station_river_creator.py

```diff
@@ -1,3 +1,124 @@
+
+            stations[reg_num] = {
+                'river_name': river_name,
+                'station_name': station_name,
+                'EOVy': self.data_handler.station_coordinates[reg_num]['EOVy'],
+                'EOVx': self.data_handler.station_coordinates[reg_num]['EOVx'],
+                'null_point': self.data_handler.station_coordinates[reg_num]['null_point']
+            }
+
+        return stations
+
+    def create_rivers(self) -> dict:
+        """
+        Function for creating the rivers data structure. A river looks like
+        {'river_name': [station1, station2, ..., stationN]}, where the stations
+        inside the list are sorted in descending order by their null points.
+        :return dict: dictionary of all rivers
+        """
+        rivers_unsorted = self.get_rivers_unsorted()
+
+        return self.sort_rivers(rivers_unsorted=rivers_unsorted)
+
+    def get_rivers_unsorted(self) -> dict:
+        """
+        Creates dictionary of dictionaries like
+        {'river_name': [station1, station2, ..., stationN]}, where the stations
+        inside the list are not sorted.
+        :return dict: dictionary of unsorted rivers
+        """
+        rivers_unsorted = {}
+        for river_name in list(self.data_handler.river_station_mapping.keys()):
+            station_name_list = self.data_handler.river_station_mapping[river_name]
+            reg_number_list = [
+                self.data_handler.station_reg_mapping[station_name] for station_name in station_name_list
+            ]
+
+            rivers_unsorted[river_name] = reg_number_list
+
+        return rivers_unsorted
+
+    def sort_rivers(self, rivers_unsorted: dict) -> dict:
+        """
+        Sorts stations in rivers in descending order by their null points
+        :param dict rivers_unsorted: dictionary of unsorted rivers
+        :return dict: dictionary of sorted rivers
+        """
+        rivers_sorted = {}
+        for river_name in list(rivers_unsorted.keys()):
+            river_unsorted = rivers_unsorted[river_name]
+            river_sorted = sorted(
+                river_unsorted,
+                key=lambda x: -self.data_handler.station_coordinates[x]['null_point']
+            )
+
+            rivers_sorted[river_name] = river_sorted
+
+        return rivers_sorted
+
+    def create_completed_rivers(self) -> dict:
+        """
+        Creates completed rivers. Uses river_connections to append the rivers with some
+        "completing" stations.
+        :return dict: dictionary of completed rivers
+        """
+        completed_rivers = copy.deepcopy(self.rivers)
+        for river_name in list(self.data_handler.river_connections.keys()):
+            close_beginning = self.data_handler.river_connections[river_name]['close_beginning']
+            close_ending = self.data_handler.river_connections[river_name]['close_ending']
+
+            if close_beginning != 0:
```

Comment:
Elegánsabb lenne a JSON-ökben "null" értéket írni, ami beolvasás után a Python-ban `None` érték lesz - tehát azt csekkolni itt, hogy `is None`.